### PR TITLE
add scala-newtype and fast-string-interpolator

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

**Comment by SethTisue** [comment]
test run queued: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2687/ ~~(404 til Jenkins finishes rousing the thunder gods from their drunken slumber)~~

**Comment by SethTisue** [comment]
try again: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2688/consoleFull

**Comment by SethTisue** [comment]
since it also came up at https://github.com/scala/scala-dev/issues/480, attempting to add fast-string-interpolator as well. fyi @Sizmek @estatico

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2689/consoleFull

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2690/consoleFull

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2691/consoleFull

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2692/consoleFull

File: configs/community.dbuild [hunk 1]
```diff
@@ -1415,6 +1415,24 @@ build += {
...
+  ${vars.base} {
+    name: "fast-string-interpolator"
+    uri:  ${vars.uris.fast-string-interpolator-uri}
+    extra.sbt-version: ${vars.sbt-1-version}
+    extra.options: ["-Dbintray.user=dummy", "-Dbintray.pass=dummy"]
+    extra.commands: ${vars.default-commands} [
+      "set every bintrayReleaseOnPublish := false"
+    ]
+    // Missing dependency: com.dongxiguo#fastring
+    extra.exclude: ["benchmark"]
```
Comment (plokhotnyuk) [inline configs/community.dbuild:1433]: I have isolated 3rd party interpolators in this module, to reproduce bug with macros compilation in `benchmark-core`: https://github.com/Sizmek/fast-string-interpolator/commit/346b61dbc3a00e623279f0faa21e307df73cc331

Comment (SethTisue) [reply configs/community.dbuild:1433]: thanks! I'll do at least one more run before merging, in order to test that change.

**Comment by SethTisue** [comment]
@plokhotnyuk at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2693/consoleFull I'm seeing

```
[fast-string-interpolator] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.11/project-builds/fast-string-interpolator-648a123d269031c17bdf894a6c53589c732c9744/benchmark-core/src/test/scala/com/sizmek/fsi/benchmark/StringConcatenationBenchmarkCoreSpec.scala:6:31: not found: type StringConcatenationBenchmark
[fast-string-interpolator] [error]   private val benchmark = new StringConcatenationBenchmark
[fast-string-interpolator] [error]                               ^
```

?

**Comment by plokhotnyuk** [comment]
Sorry, have fixed it by https://github.com/Sizmek/fast-string-interpolator/commit/a961515c49052dec51860cf2fb335048d232e77a

It looks like cross-module dependencies are allowed, while wasn't declared... Should I report it to SBT team?

**Comment by SethTisue** [comment]
green at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2694/console

**Comment by SethTisue** [comment]
👍 thank you for your early testing of 2.12.5!

**Pull request merged**